lib/Target/SubstraitPB/Import.cpp

```diff
@@ -155,16 +176,18 @@ importSetRel(ImplicitLocOpBuilder builder, const Rel &message) {
   if (failed(leftOp) || failed(rightOp))
     return failure();
 
-  if (setRel.op() != SetRel::SET_OP_UNION_DISTINCT)
-    return mlir::emitError(
-        builder.getLoc(),
-        "only 'union distinct' set operation supported for import");
+  std::optional<SetOpKind> kind = getSetOpKind(setRel.op());
+
+  // Check for unsupported set operations.
+  if (!kind)
+    return mlir::emitError(builder.getLoc(),
+                           "Set operation not supported for import");
```

Comment:
It is probably wise to make this a compiler failure rather than an assertion because it could be that the protobuf definitions get extended (and maybe it's even possible to have invalid values in protobuf). But I would change the message to something like "unexpected 'operation' found". If possible, it'd be great to specify the value as well.

Reply:
My impression was that returning the result of emitError would return failure?

Reply:
Yes, yes, when I said "it is probably wise", I was confirming that you took the right decision. Sorry about the confusion.